src/classes/CRLP_Rollup_SEL.cls

```diff
@@ -282,17 +367,22 @@ public class CRLP_Rollup_SEL {
 
         // Determine the correct base summary object based on the Rollup Type
         SObjectType summaryObjectType;
-        if (rollupType == CRLP_RollupProcessingOptions.RollupType.AccountHardCredit ||
-                rollupType == CRLP_RollupProcessingOptions.RollupType.AccountContactSoftCredit ||
-                rollupType == CRLP_RollupProcessingOptions.RollupType.AccountSoftCredit) {
-            summaryObjectType = Account.SObjectType;
-        } else if (rollupType == CRLP_RollupProcessingOptions.RollupType.ContactHardCredit ||
-                rollupType == CRLP_RollupProcessingOptions.RollupType.ContactSoftCredit) {
-            summaryObjectType = Contact.SObjectType;
-        } else if (rollupType == CRLP_RollupProcessingOptions.RollupType.GAU) {
-            summaryObjectType = General_Accounting_Unit__c.SObjectType;
-        } else if (rollupType == CRLP_RollupProcessingOptions.RollupType.RecurringDonations) {
-            summaryObjectType = npe03__Recurring_Donation__c.SObjectType;
+        switch on (rollupType) {
+            when AccountHardCredit, AccountContactSoftCredit, AccountSoftCredit {
+                summaryObjectType = Account.SObjectType;
+            }
+
+            when ContactHardCredit, ContactSoftCredit {
+                summaryObjectType = Contact.SObjectType;
+            }
+
+            when GAU {
+                summaryObjectType = General_Accounting_Unit__c.SObjectType;
+            }
+
+            when RecurringDonations {
+                summaryObjectType = npe03__Recurring_Donation__c.SObjectType;
+            }
         }
 
         // Retrieve all the Rollup Definitions for the summary object
```

Comment:
How about removing line 390 and extracting lines 391 to 436  into a new method `shouldIncludeRollup()`. So if we extract `summarySObjectType` code into a `determineSummarySObjectType()`, we would have : 

```
    //lines 368 - 440 replacement
    SObjectType summarySObjectType = determineSummarySObjectType(rollupType);

    List<Rollup__mdt> rollups = CRLP_Rollup_SEL.getRollupsBySummaryType().get(summaryObjectType);
    rollups = rollups == null ? new List<Rollup__mdt>() : rollups;

    for (Rollup__mdt rollup : rollups) {
        if (shouldIncludeRollup(rollup, rollupType, typeFilter)) {
            filteredRollups.add(rollup);
        }
    }

    // Save the list in the cache
    cachedRollupsByTypeAndFilter.put(rollupType.name() + typeFilter.name(), filteredRollups);
    return filteredRollups;
```

Reply:
Great suggestion!